### ISSUE TITLE
Paste as JPG

### ISSUE THREAD

**Issue by Laronski** [OpenLiveWriter/OpenLiveWriter#138]
Open Live Writer needs to have the ability to paste as JPG with the option to optimize the image for best image size ratio. Hopefully, plug-ins can handle the image editing post-paste.

**Comment by ScottIsAFool** [comment]
I have a plugin called Clipboard Capture that does [almost] exactly that. Been around for Live Writer for a long time now. The current version won't work with OLW as it is, but that will shortly be changing. Keep your eyes open. 

That said, having this kind of functionality built in totally makes sense! @RobDolinMS

**Comment by Laronski** [comment]
Yes; I used to have that. I use Sharper Photo now, but yes; having it in the application would be very nice.

**Comment by ScottIsAFool** [comment]
Is that a Plugin? If so, got a link to it?

**Comment by Laronski** [comment]
http://windows-live-writer-plugin-sharper-photo.software.informer.com/

It's very nice. Offers paste as JPG, but also editing images after the paste; before publishing. You'll like it. This functionality needs to be in Open Live Writer.

**Comment by martinwoodward** [comment]
Would it be better to have paste as PNG by default?  Graphics are often screenshots so a lossless format would be neat (it's what GitHub does when you paste an image into an issue etc)

**Comment by Laronski** [comment]
PNG produces larger image files than JPG and Windows Live Writer already pasted images as PNG; thus the need to paste as JPG. If you don't care about bandwidth, site storage and blog loading times; PNG is the right choice.

**Comment by vhanla** [comment]
Replacing `image.png` with `image.jpg` in `OpenLiveWriter.PostEditor.PostHtmlEditing.ImageClipboardFormatHandler` `SaveImageDataToTempFile` will make it save as jpg, since DIBHelper calls its `GetCodecClsid` function to find and set the proper codec according to filename extension. However, it might be better to add options to choose among file formats, maybe inside Editing section in Options dialog as well the compression level. 

Just to test it, I renamed it to .JPG extension as follows:
```cs
            else if (imageData.Dib != null)
            {
                //save the DIB data as a PNG
                savedImagePath = TempFileManager.Instance.CreateTempFile("image.jpg");
``` 
And that was enough (at least for me) and it saved the picture from clipboard as a JPG file with the default compression level. I have tested with GreenShot's clipboard format which is DIB, but Bitmap format is another function in the same file. 
Notice, though, that image editing (shadow, effects, watermark, etc.) should also consider compression level and the file format. However, it must use bitmap or png file as source image before returning the jpg result image, otherwise the original picture would lose quality if jpg format is used as the original image source due to its lossy feature.

[UPDATE] I did some coding in another branch at my fork which allows to paste it as JPEG or PNG and also you can set the compression level. However, compression level is not working with picture effects, watermark, etc. it only works with paste from clipboard. 

So far, this is how it looks in settings panel:
![img](https://lh3.googleusercontent.com/9SpxFiqx8PB8GtdZpACzytv72QqF1y8UqCnjYBpq1lS_yzKDItdiCa0QKjvLlScmP38l6QP58rlawQrCxL-CjYfUt6qSGmrN=w1366-h768-rw-no)

Here you can find this branch (WIP) https://github.com/vhanla/OpenLiveWriter/tree/PasteAsJpg
And these are the diffs https://github.com/vhanla/OpenLiveWriter/commit/21711cf14e09c16c2385da4a21e181bef222c125

**Comment by shanselman** [comment]
Thanks for reporting this and for helping Open Live Writer over the years. We are doing an issue-bankruptcy cleanup so we can make the repository maintainable again and focus on current, actionable work. This older issue is being closed as part of that cleanup, not because it was unimportant. If this still reproduces in the current code or is still relevant to the revitalization work, please open a fresh issue with current details and we will triage it going forward.
